### PR TITLE
MM-54989 Fix team run team interaction.

### PR DESCRIPTION
**Pull request by crspeller**
#### Summary

Adding checks for team permissions when accessing runs. 

#### Ticket

https://mattermost.atlassian.net/browse/MM-54989

### CODE REVIEW

**Review by lieut-data — APPROVED** [review]

File: server/api_runs_test.go [hunk 1]
```diff
@@ -548,6 +548,15 @@ func TestRunPostStatusUpdate(t *testing.T) {
 		assert.Error(t, err)
 	})
 
+	t.Run("no permissions to run", func(t *testing.T) {
+		_, err := e.ServerAdminClient.RemoveTeamMember(e.BasicRun.TeamID, e.RegularUser.Id)
+		require.NoError(t, err)
+		err = e.PlaybooksClient.PlaybookRuns.UpdateStatus(context.Background(), e.BasicRun.ID, "update", 600)
```
Comment (esarafianou) [inline server/api_runs_test.go:554]: UpdateStatus doesn't call the `POST /plugins/playbooks/api/v0/runs/<RUN_ID>/update-status-dialog` endpoint. It calls the `runs/<RUN_ID>/status` endpoint which is not what we'd like to test: https://github.com/mattermost/mattermost-plugin-playbooks/blob/a10bb42418667c0f5337f44b39e582f7a8fc81cf/client/playbook_runs.go#L141C27-L141C43.

File: server/app/permissions_service.go [hunk 2]
```diff
@@ -410,6 +410,10 @@ func (p *PermissionsService) RunManageProperties(userID, runID string) error {
 }
 
 func (p *PermissionsService) runManagePropertiesWithPlaybookRun(userID string, run *PlaybookRun) error {
```
Comment (esarafianou) [inline server/app/permissions_service.go:412]: The `POST /plugins/playbooks/api/v0/runs/<RUN_ID>/update-status-dialog` doesn't seem to ever call the `runManagePropertiesWithPlaybookRun` function so this fix won't get applied in that endpoint: https://github.com/mattermost/mattermost-plugin-playbooks/blob/a10bb42418667c0f5337f44b39e582f7a8fc81cf/server/api/playbook_runs.go#L922-L971

**Review by esarafianou — CHANGES_REQUESTED** [review]

File: server/app/permissions_service.go (hunk 2 above)
Comment (crspeller) [reply server/app/permissions_service.go:412]: It is a little confusing if you don't know what is going on. There is actually some registered middleware that is taking care of it. You can see it here: https://github.com/mattermost/mattermost-plugin-playbooks/blob/d03dfd8d4e200e8db5c5f64ae05a7ba4b68032fc/server/api/playbook_runs.go#L79

File: server/api_runs_test.go (hunk 1 above)
Comment (crspeller) [reply server/api_runs_test.go:603]: The code paths for checking permissions are identical. The only difference is the input.
However I added a test for it just to be safe.

**Review by sbishel — APPROVED** [review]

File: server/app/permissions_service.go (hunk 2 above)
Comment (esarafianou) [reply server/app/permissions_service.go:412]: Aah, that makes sense. Thanks for pointing to that! I missed it.

**Review by esarafianou — APPROVED** [review]
LGTM!

**Pull request merged**